class/practice2.2/main.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <sstream>
+#include <algorithm>
+
+using namespace std;
+
+struct Pixel {
+    int r, g, b;
+};
+
+bool isValidPixel(const Pixel &pixel) {
+    return (pixel.r >= 0 && pixel.r <= 255) && (pixel.g >= 0 && pixel.g <= 255) && (pixel.b >= 0 && pixel.b <= 255);
+}
+
+bool loadPixels(const string &filename, vector<vector<Pixel>> &pixels, string &errorMessage) {
```

Comment:
In your functions you have output argument pixels. It is a signal that it's time to encapsulate the things.
Let's introduce an Image class that will have std::vector<std::vector<Pixel>> as a private member. loadPixels and other functions can be then rewritten as methods of that class
```
class {

     bool loadFromFile(const std::string& filename); // std::error_code  or std::expected can be used as a return type to error message if needed.
    
private:
     std::vector<std::vector<Pixel>> data;
};
```